lib/ansible/modules/cloud/scaleway/scaleway_snapshots_facts.py

```diff
@@ -0,0 +1,133 @@
+    - "This module gather facts about snapshots on Scaleway."
+options:
+
+  oauth_token:
+    description:
+     - Scaleway OAuth token.
+    required: true
+
+  region:
+    description:
+    - Scaleway compute zone
+    required: true
+    choices:
+      - ams1
+      - EMEA-NL-EVS
+      - par1
+      - EMEA-FR-PAR1
+
+  timeout:
+    description:
+    - Timeout for API calls
+    required: false
+    default: 30
+'''
+
+EXAMPLES = '''
+- name: Get snapshots information
+  scaleway_snapshots_facts:
+    region: par1
+'''
+
+RETURN = '''
+data:
+    description: This is only present when C(state=present)
+    returned: when C(state=present)
+    type: dict
+    sample: {
+        "snapshots": [
+            {
+              "state": "available",
+              "base_volume": null,
+              "name": "odoo-snapshot",
+              "modification_date": "2018-06-18T11:24:46.272833+00:00",
+              "organization": "951df375-e094-4d26-97c1-ba548eeb9c42",
+              "size": 50000000000,
+              "id": "c18dd10e-86c2-443d-81d9-1ecbdc959e3b",
+              "volume_type": "l_ssd",
+              "creation_date": "2018-06-06T08:40:45.395451+00:00"
+            },
+            {
+              "state": "available",
+              "base_volume": null,
+              "name": "neo4j-genealogy-snapshot",
+              "modification_date": "2018-05-21T10:19:03.868987+00:00",
+              "organization": "951df375-e094-4d26-97c1-ba548eeb9c42",
+              "size": 50000000000,
+              "id": "93bf81ec-f778-47ca-a6f3-9c320f646464",
+              "volume_type": "l_ssd",
+              "creation_date": "2018-05-03T13:58:46.416778+00:00"
+            }
+        ]
+    }
+'''
+
+from ansible.module_utils.basic import AnsibleModule, env_fallback
+from ansible.module_utils.scaleway import ScalewayAPI, SCALEWAY_LOCATION
+
+
+def core(module):
+    api_token = module.params['oauth_token']
+    region = module.params["region"]
+
+    compute_api = ScalewayAPI(module=module,
+                              headers={'X-Auth-Token': api_token},
+                              base_url=SCALEWAY_LOCATION[region]["api_endpoint"])
+
+    response = compute_api.get("snapshots")
+    status_code = response.status_code
+
+    if not response.ok:
+        module.fail_json(msg='Error fetching facts [{0}: {1}]'.format(
+            status_code, response.json['message']))
+
+    snapshots = response.json["snapshots"]
+    module.exit_json(changed=False, snapshots=snapshots)
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            oauth_token=dict(
+                no_log=True,
+                # Support environment variable for Scaleway OAuth Token
+                fallback=(env_fallback, ['SCW_TOKEN', 'SCW_API_KEY', 'SCW_OAUTH_TOKEN']),
+                required=True,
+            ),
+            region=dict(required=True, choices=SCALEWAY_LOCATION.keys()),
+            timeout=dict(type="int", default=30),
```

Comment:
`̀timeout` parameter is used by `module_utils.scaleway.ScalewayAPI.send`, it would be better to move `timeout` definition in `module_utils/scaleway.py` (something like [that](https://github.com/ansible/ansible/blob/583e5b4a16f791bf8fb8254fe1439a22ec4a13f5/lib/ansible/module_utils/cloudstack.py#L37)), then this parameter documentation could be factorized too ([example](https://github.com/ansible/ansible/blob/583e5b4a16f791bf8fb8254fe1439a22ec4a13f5/lib/ansible/utils/module_docs_fragments/cloudstack.py))

Reply:
I think that this is what @Spredzy is doing in #43762 at least for the documentation part. @Spredzy do you have complete examples of the kind of module you mention?

Reply:
@sieben yes, this is what we do in the Vultr modules. https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/vultr.py

Basically, there is a base class located in utils and all the modules extends it and bring only the module specifics in it. #43762 is only for documenation. I haven't worked on a base class, but I can start working on it.